### PR TITLE
Finished task, try-1.

### PR DESCRIPTION
**Pull request by slava-moiseev-kramatorsk**

### CODE REVIEW

**Comment by slava-moiseev-kramatorsk** [comment]
Результат покриття
image

File: checklist.md [hunk 1]
```diff
@@ -31,17 +31,17 @@ Same goes for files that you use in tests, **let's put them into this folder:**
 #### Try to cover different scenarios in tests
 Your task is to include edge cases apart from the regular method use case.
 
-#### Don't test the Main class
-We want to test only business logic, so there is no need to cover the `Main` class with tests. 
-You can exclude the Main class with the `main()` method from being checked for code coverage in `pom.xml`.   
+#### Don't test the main class
```
Comment (okuzan) [inline checklist.md:34]: always review what you're committing, to avoid unintended changes, be careful with project-wide replacements

File: src/main/java/files/InputData.CSV [hunk 2]
```diff
@@ -0,0 +1,15 @@
+type,fruit,quantity
```
Comment (okuzan) [inline src/main/java/files/InputData.CSV:1]: we use src/main/resources , src/test/resources for this

File: src/test/java/impl/FileReaderImplTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.List;
+import org.junit.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class FileReaderImplTest {
+    private static final String PATH = "src/test/java/TestsFiles/TestInputData.csv";
+
+    @Test
+    public void reader_Test_Ok() {
```
Comment (okuzan) [inline src/test/java/impl/FileReaderImplTest.java:14]: be consistent in naming, always use lowercase after underscore
first part is name of method under test
second part of the name describes input, arguments, conditions, what you are passing to the tested function
```suggestion
    public void readFile_existingFile_ok() {
```

File: src/test/java/impl/FileReaderImplTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                "s,apple,10",
+                "s,banana,100",
+                "p,apple,10",
+                "p,apple,20",
+                "r,banana,10",
+                "r,apple,10"
+        };
+        List<String> expectedData = List.of(expectedDataArray);
+        assertEquals(reader.readFile(PATH), expectedData,
+                "Your FileReaderImpl class work incorrectly");
```
Comment (okuzan) [inline src/test/java/impl/FileReaderImplTest.java:28]:
```suggestion
                "Your FileReaderImpl class works incorrectly");
```

File: src/test/java/impl/ParseDataServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            new FruitTransaction("p", "apple", 20),
+            new FruitTransaction("r", "banana", 10),
+            new FruitTransaction("r", "apple", 10)
+    };
+    private final ParseDataServiceImpl parse = new ParseDataServiceImpl();
+    private final FileReaderImpl reader = new FileReaderImpl();
+
+    @BeforeAll
+    void setUp() {
+    }
```
Comment (okuzan) [inline src/test/java/impl/ParseDataServiceImplTest.java:29]:
```suggestion

File: src/test/java/impl/ParseDataServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Test
+    public void parseData_validData_Ok() {
+        String path = "src/test/java/TestsFiles/TestInputData.csv";
+        List<FruitTransaction> actualData = parse.parseData(reader.readFile(path));
+        List<FruitTransaction> expectedData = List.of(TestsTransaction);
+        assertEquals(expectedData, actualData,
+                "Your ParseService class work incorrectly");
+
```
Comment (okuzan) [inline src/test/java/impl/ParseDataServiceImplTest.java:38]:
```suggestion
```

File: src/test/java/impl/ParseDataServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import model.FruitTransaction;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ParseDataServiceImplTest {
+    private static final FruitTransaction[] TestsTransaction = new FruitTransaction[] {
```
Comment (okuzan) [inline src/test/java/impl/ParseDataServiceImplTest.java:14]:
```suggestion
    private static final FruitTransaction[] testTransactions = new FruitTransaction[] {
```
don't capitalize var's name

File: src/test/java/impl/ParseDataServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    private final ParseDataServiceImpl parse = new ParseDataServiceImpl();
+    private final FileReaderImpl reader = new FileReaderImpl();
+
+    @BeforeAll
+    void setUp() {
+    }
+
+    @Test
+    public void parseData_validData_Ok() {
+        String path = "src/test/java/TestsFiles/TestInputData.csv";
```
Comment (okuzan) [inline src/test/java/impl/ParseDataServiceImplTest.java:33]: extract to constant

File: src/test/java/impl/ReportServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        reportService = new ReportServiceImpl();
+        transaction = new TransactionProcessor();
+
+        List<String> dataFromFile = fileReader.readFile(PATH);
+        List<FruitTransaction> fruitsList = parse.parseData(dataFromFile);
+        transaction.transactionFruits(fruitsList);
+    }
+
+    @Test
+    void createReport_0k() {
```
Comment (okuzan) [inline src/test/java/impl/ReportServiceImplTest.java:35]: ok, not 0k  :))
please fix naming everywhere according to the test naming convention described in task description

File: src/test/java/impl/WriterServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        writeService.writeDataToFile(expectedResult, PATH);
+        StringBuilder testDataFromFile = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(PATH))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                testDataFromFile.append(line).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file by path: " + PATH, e);
+        }
```
Comment (okuzan) [inline src/test/java/impl/WriterServiceImplTest.java:43]: extract to a separate method

File: src/test/java/service/OperationTest.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+import model.FruitTransaction;
+import org.junit.jupiter.api.Test;
+import strategy.Operation;
+
+class OperationTest {
+    private static final FruitTransaction validFruit =
+            new FruitTransaction("b", "banana", 100);
+    private static final FruitTransaction invalidOperationField =
+            new FruitTransaction("q", "banana", 100);
```
Comment (okuzan) [inline src/test/java/service/OperationTest.java:14]: we don't test models, they are considered tested implicitly when we test services that use them

File: src/test/java/storage/StorageTest.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.jupiter.api.TestInstance;
+import service.TransactionProcessor;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class StorageTest {
+    private static FileReaderImpl fileReader;
+    private static ParseDataServiceImpl parse;
+    private static ReportServiceImpl reportService;
+    private static WriterServiceImpl writerService;
+    private static TransactionProcessor transaction;
```
Comment (okuzan) [inline src/test/java/storage/StorageTest.java:22]: you included real implementations of all your services for storage unit testing, when we test some class, we test only this class and mock other dependencies. We can't include our other classes, which can be faulty too
storage test isn't required, you can remove it altogether

**Review by okuzan — CHANGES_REQUESTED** [review]
please review your naming for each test method
add test for retrieving a correct handler for a given operation type, and for handlers themselves

**Comment by slava-moiseev-kramatorsk** [comment]
> please review your naming for each test method add test for retrieving a correct handler for a given operation type, and for handlers themselves

Так є тест  -> service/OperationTest.  Чи мається на увазі інше ?

File: src/main/java/files/InputData.CSV (hunk 2 above)
Comment (slava-moiseev-kramatorsk) [reply src/main/java/files/InputData.CSV:1]: Змінив

File: src/test/java/impl/ParseDataServiceImplTest.java (hunk 5 above)
Comment (slava-moiseev-kramatorsk) [reply src/test/java/impl/ParseDataServiceImplTest.java]: прибрав

File: src/test/java/impl/ReportServiceImplTest.java (hunk 9 above)
Comment (slava-moiseev-kramatorsk) [reply src/test/java/impl/ReportServiceImplTest.java]: пофіксив

File: src/test/java/impl/WriterServiceImplTest.java (hunk 10 above)
Comment (slava-moiseev-kramatorsk) [reply src/test/java/impl/WriterServiceImplTest.java:49]: зробив

File: src/test/java/storage/StorageTest.java (hunk 12 above)
Comment (slava-moiseev-kramatorsk) [reply src/test/java/storage/StorageTest.java]: прибрав

**Review by kozhukhovsky — APPROVED** [review]
